api/protocol.go

```diff
@@ -52,68 +50,148 @@ type Response struct {
 	Data    map[string]interface{} `json:"data,omitempty"`
 }
 
-// ReadMessage reads a message from reader. A message is either a Request or a
-// Response
-func ReadMessage(reader io.Reader, msg interface{}) error {
-	buf := make([]byte, headerLength)
-	n, err := reader.Read(buf)
+func maxOpcodeForFrameType(t FrameType) int {
+	switch t {
+	default:
+		fallthrough
```

Comment:
I don't understand this - if you `fallthrough` here, shouldn't you also `fallthrough` in `TypeCommand` too, so that all 3 values return `int(CmdMax)`?